[scan] Add option to specify JUnit report output file name

PycKamil opened the pull request:
Scan junit report file had .junit extension, so it was not possible to use with some CI, fe. Bamboo. 

It was bugging me for a long time, in the end it was the reason to write my first line of ruby code ever. Hope it didn't breake anything 😁 

This solves #1942

PycKamil committed:
[scan] Changed JUnit report output type

Changed JUnit report output file type to xml. This solves #1942

KrauseFx commented on scan/lib/scan/report_collector.rb:
```diff
@@ -56,6 +56,8 @@ def generate_commands(path, types: nil, output_file_name: nil)
       if @use_clang_report_name && type == "json-compilation-database"
         "compile_commands.json"
+      elsif type == "junit"
+        "report.xml"
```
This will break all existing setups that look for a `*.junit` file in their build output.

PycKamil replied:
You're right... I will try to solve it by making new option so it won't break existing setups.

PycKamil committed:
[scan] Added new option for defining custom report file extension

Instead of chaning junit type extension, added optional
'custom_report_extension' that will not impact curent users

PycKamil committed:
[scan] Added new configuration option

code clean up

PycKamil committed:
[scan] added unit tests

PycKamil committed:
[scan] fixed detected code offense

PycKamil commented:
I have created new parameter for setting explicitly report file extension.@KrauseFx what do you think about this solution?

KrauseFx commented:
Great, thanks for updating the PR, this works for me, let's get another set of 👀 before merging 👍

mfurtak commented:
This seems OK to me, but if we want to allow specifying a custom file extension, does it make sense to just allow control over the full report output file name instead? That way, if you want **report.xml** you can specify that, but if you want **special.xml**, that's possible too?

PycKamil committed:
[scan] Changed specifing report extension to full report output file name instead

PycKamil commented:
@mfurtak good idea, there could be some setups when you have multiple target and tests so being able to have unique reports name should be useful. I have changed my pull request to setting full report name.

KrauseFx commented:
Great feedback @mfurtak and thanks for updating the PR @PycKamil 👍

mfurtak commented:
Awesome! Thanks for putting this together, @PycKamil 🚀
